src/rhub/api/tower.py

```diff
@@ -282,3 +283,60 @@ def get_job_stdout(job_id):
     except Exception as e:
         logger.exception(e)
         return problem(500, 'Server Error', 'Uknown server error, {e}')
+
+
+def webhook_auth(username, password, required_scopes=None):
+    # Tower offers sending 'basic' auth credentials to protect access
+    # to the webhook_notification() endpoint.  The credentials are
+    # defined/stored in Vault.
+
+    try:
+        if (username == current_app.config['WEBHOOK_USER']
+                and password == current_app.config['WEBHOOK_PASS']):
+            return {'sub': 'webhook'}    # successful authentication
+
+    except KeyError as e:
+        logger.exception('Missing tower webhook notification credential(s)'
+                         f' {e}; notification ignored')
+
+    logger.warning('Incorrect tower webhook notification credentials supplied;'
+                   ' notification ignored')
+
+    raise Unauthorized('Incorrect tower webhook notification credentials'
+                       ' supplied')
+
+    return None    # failed authentication
```

Comment:
dead code. `return` can never be executed because of `raise` above

Reply:
thanks!  totally missed that!